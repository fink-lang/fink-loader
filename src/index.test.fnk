{describe, it, expect, to_match_snapshot, mock_fn} = import '@fink/jest/test.fnk'

{loader} = import './index.fnk'


mock_callback = mock_fn _

ctx = dict:
  query: ''
  async: mock_fn fn: mock_callback
  resourcePath: './test.fnk'


describe loader, fn:

  it 'compiles fink', fn:
    await loader
      ctx
      'foo = 1234 or spam'

    [[,code]] = mock_callback.mock.calls

    expect
      code
      to_match_snapshot


  it 'compiles with sourcemap', fn:
    await loader
      {...ctx, query: '?{"source_maps": true}'}
      'foo = 1234 or spam'

    [[,, source_map]] = mock_callback.mock.calls

    expect
      source_map
      to_match_snapshot


  it 'handles parsing errors', fn:
    await loader
      ctx
      'foo = ['

    [[err]] = mock_callback.mock.calls

    expect
      err
      to_match_snapshot


  it 'handles compile errors', fn:
    await loader
      ctx
      '123 = foo'

    [[err]] = mock_callback.mock.calls

    expect
      err
      to_match_snapshot

