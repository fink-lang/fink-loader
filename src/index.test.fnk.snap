// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader compiles fink 1`] = `"export const foo = 1234 || spam;"`;

exports[`loader compiles with sourcemap 1`] = `
Object {
  "mappings": "AAAA,aAAA,GAAG,GAAG,QAAQ,IAAd",
  "names": Array [],
  "sources": Array [
    "./test.fnk",
  ],
  "sourcesContent": Array [
    "foo = 1234 or spam",
  ],
  "version": 3,
}
`;

exports[`loader handles compile errors 1`] = `
[Error: ./test.fnk:1:0
1| 123 = foo
   ^

Unable to transform \`assign =\`.

TypeError: Property left of AssignmentExpression expected node to be of a type ["LVal"] but instead got "NumericLiteral"
]
`;

exports[`loader handles parsing errors 1`] = `
[Error: ./test.fnk:1:7
1| foo = [
          ^

Expected \`,\` or indented(>=1) new line or \`]\`.
]
`;
